Issue 45498: Show cross-linked peptide sequence info

labkey-jeckels opened the pull request:
#### Rationale
To better support multi attribute method analysis, Skyline added support for cross-linked peptides a couple of years ago. These are handled as special peptide modification annotations. Panorama can show the string representation but would greatly benefit from a more intuitive rendering, where we show all of the peptides involved in the linking and highlight the amino acids involved in the linking. We also want to improve the peptide mapping report for MAM folders to show more details about how and where the peptides are linked.

#### Changes
* Parse cross linked peptide sequences into a reusable data structure
* Render all peptides with link locations as modifications
* Unit tests for parsing, Selenium tests for peptide map rendering
* New/updated columns in Peptide Map report for proteins involved in linking and location of links
* More flexible column filtering for custom, MAM-related PTM report

labkey-jeckels committed:
Issue 45498: Show cross-linked peptide sequence info

ankurjuneja commented on src/org/labkey/targetedms/query/SamplePivotCustomizer.java:
```diff
@@ -33,92 +31,157 @@
 
-    @Override
-    public void customize(TableInfo tableInfo)
+    private Integer getRunId()
```
As runIds are of type long now, better to be consistent across.

labkey-jeckels replied:
We're a long way from 2 billion runs uploaded but agree best to parse as the right data type. Updated

ankurjuneja commented on src/org/labkey/targetedms/query/SamplePivotCustomizer.java:
```diff
@@ -33,92 +31,157 @@
-                filterURL = settings.getSortFilterURL();
+                try
+                {
+                    // Get the sample names associated with that run
```
I am confused with the comment here.

labkey-jeckels replied:
Fixed. Agreed that it didn't make sense where it landed after the refactor

labkey-jeckels committed:
Code review feedback

labkey-jeckels committed:
Merge branch 'release22.3-SNAPSHOT' into 22.3_fb_mamReporting

labkey-jeckels committed:
Verify peptide map report is correct for cross-linked peptides

Handle crosslinked peptides that don't specify all non-linked amino acids

Export formatted version of crosslinks

labkey-jeckels committed:
Update to match new HTML

labkey-jeckels committed:
Fix string formatting

labkey-jeckels committed:
Show molecule labels in comparison plots even without custom ion names
